#!/usr/bin/env python
PACKAGE = "moveit_planners_ompl"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("simplify_solutions", bool_t, 1,
        "Flag indicating whether computed motion plans are also simplified", True)
gen.add("minimum_waypoint_count", int_t, 2,
        "Set the minimum number of waypoints to include in a motion plan", 10, 2, 10000)
gen.add("max_solution_segment_length", double_t, 3,
        "The maximum distance between consecutive waypoints along the solution path (0.0 means 'ignore')", 0.0, 0.0, 50.0)
gen.add("use_constraints_approximations", bool_t, 4,
        "Flag indicating whether constraint approximations should be used if available.", True)
gen.add("max_planning_threads", int_t, 5,
        "The maximum number of threads to use for planning.", 4, 1, 64)
gen.add("max_goal_samples", int_t, 6,
        "The maximum number of goal samples to consider.", 10, 1, 10000)
gen.add("max_state_sampling_attempts", int_t, 7,
        "The maximum number of attempts to use when sampling a new state.", 4, 1, 10000)
gen.add("max_goal_sampling_attempts", int_t, 8,
        "The maximum number of attempts to sample a new goal.", 1000, 1, 10000)

exit(gen.generate(PACKAGE, PACKAGE, "OMPLDynamicReconfigure"))
